src/Cake.Core/Scripting/Processors/ToolDirectiveProcessor.cs

```diff
@@ -0,0 +1,191 @@
+{
+    /// <summary>
+    /// Processor for #tool directives.
+    /// </summary>
+    public sealed class ToolDirectiveProcessor : LineProcessor
+    {
+        private static FilePath _nugetPath;
+        private readonly IFileSystem _fileSystem;
+        private readonly ICakeEnvironment _environment;
+        private readonly ICakeLog _log;
+        private readonly IToolResolver _nugetToolResolver;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ToolDirectiveProcessor" /> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="log">The log.</param>
+        /// <param name="nugetToolResolver">The NuGet tool resolver.</param>
+        public ToolDirectiveProcessor(IFileSystem fileSystem, ICakeEnvironment environment, ICakeLog log, INuGetToolResolver nugetToolResolver)
+            : base(environment)
+        {
+            if (fileSystem == null)
+            {
+                throw new ArgumentNullException("fileSystem");
+            }
+            if (environment == null)
+            {
+                throw new ArgumentNullException("environment");
+            }
+            if (log == null)
+            {
+                throw new ArgumentNullException("log");
+            }
+            if (nugetToolResolver == null)
+            {
+                throw new ArgumentNullException("nugetToolResolver");
+            }
+
+            _fileSystem = fileSystem;
+            _environment = environment;
+            _log = log;
+            _nugetToolResolver = nugetToolResolver;
+        }
+
+        /// <summary>
+        /// Processes the specified line.
+        /// </summary>
+        /// <param name="processor">The script processor.</param>
+        /// <param name="context">The script processor context.</param>
+        /// <param name="currentScriptPath">The current script path.</param>
+        /// <param name="line">The line to process.</param>
+        /// <returns>
+        ///   <c>true</c> if the processor handled the line; otherwise <c>false</c>.
+        /// </returns>
+        public override bool Process(IScriptProcessor processor, ScriptProcessorContext context, FilePath currentScriptPath, string line)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            var tokens = Split(line);
+            var directive = tokens.FirstOrDefault();
+            if (string.IsNullOrWhiteSpace(directive))
+            {
+                return false;
+            }
+
+            if (!directive.Equals("#tool", StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            // Fetch the tool NuGet ID.
+            var toolId = tokens
+                .Select(value => value.UnQuote())
+                .Skip(1).FirstOrDefault();
```

Comment:
What happens if someone write `#tool" with nothing else on the line? Will the `Skip` throw? If so, there should probably be a split and a count here to give the user a good error message.